velox/vector/BaseVector.h

```diff
@@ -177,6 +177,14 @@ class BaseVector {
     return type_;
   }
 
+  void setType(TypePtr type) {
+    VELOX_CHECK_NOT_NULL(type);
+    VELOX_CHECK(
```

Comment:
@mbasmanova We have checked that the type's kind is equal to 'kindEquals' with the given type. Therefore, there won't be any instances where INTEGER() is converted to DATE() or VARCHAR is converted to REAL, right?
https://github.com/facebookincubator/velox/pull/6074#discussion_r1407470529